### PR TITLE
win: Add appropriate flags for Clang-cl

### PR DESCRIPTION
**Pull request by tdusnoki**

### CODE REVIEW

**Review by pawosm-arm — APPROVED** [review]
I don't owe any of the Microsoft compilers to verify that, so I trust you it was written correctly.

File: CMakeLists.txt [hunk 1]
```diff
@@ -283,17 +283,21 @@ else()
...
-     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-error=sign-compare -Wno-error=pointer-sign -Wno-error=incompatible-pointer-types -Wno-error=string-conversion -Wno-error=c99-extensions -Wno-error=deprecated-register -Wno-error=format -Wno-error=format-pedantic -Wno-error=tautological-constant-out-of-range-compare -Wno-error=switch -Wno-error=covered-switch-default -Wno-error=enum-compare -Wno-error=char-subscripts -Wno-error=parentheses -Wno-error=array-bounds -Wno-error=fortify-source -Wno-error=comment")
-     if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION GREATER_EQUAL 13)
-       set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -Wno-error=unused-but-set-variable")
-       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=unused-but-set-variable")
-     endif()
+  set(WITH_WERROR ON)
```
Comment (isuruf) [inline CMakeLists.txt:290]: This makes it impossible to turn off `WITH_ERROR` option using cmake command line. Please delete Line 286 and bring back Line 289

**Review by isuruf — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (tdusnoki) [reply CMakeLists.txt]: Good catch, thanks!

**Review by isuruf — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -287,13 +287,16 @@ endif()
...
-     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wno-error=sign-compare -Wno-error=pointer-sign -Wno-error=incompatible-pointer-types -Wno-error=string-conversion -Wno-error=c99-extensions -Wno-error=deprecated-register -Wno-error=format -Wno-error=format-pedantic -Wno-error=tautological-constant-out-of-range-compare -Wno-error=switch -Wno-error=covered-switch-default -Wno-error=enum-compare -Wno-error=char-subscripts -Wno-error=parentheses -Wno-error=array-bounds -Wno-error=fortify-source -Wno-error=comment")
-     if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION GREATER_EQUAL 13)
-       set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -Wno-error=unused-but-set-variable")
-       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=unused-but-set-variable")
-     endif()
+elseif (MSVC)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /std:c++14 -Woverloaded-virtual -Wno-long-long -Wno-unused-parameter -Wwrite-strings")
+  option(WITH_WERROR "Compile with '-Werror' C/C++ compiler flag" OFF)
```
Comment (bryanpkc) [inline CMakeLists.txt:292]: Unless MSVC supports GCC-style `-W` options, I think this option definition allows someone to run a build on Windows with `-DWITH_WERROR=ON` and end up with a lot of "unsupported option" errors.

**Comment by Neumann-A** [comment]
Hmm does this fix the compiler errors due to e.g. `m256i` casts to `m256` ? I feel like libpgmath should use the appropriate intrinsics but if just removing some flags helps i am all in.

File: CMakeLists.txt (hunk 2 above)
Comment (tdusnoki) [reply CMakeLists.txt]: According to my tests MSVC supports these warning options and with `-DWITH_WERROR=ON` I got no error messages on Windows.

**Comment by kaadam** [comment]
@bryanpkc Hi, Could you take a look at it again please? Thanks.

**Review by bryanpkc — APPROVED** [review]

**Pull request merged**